testing/test_script.py

```diff
@@ -1,18 +1,250 @@
+    subprocess.check_output(command, shell=True)
+
+    list1 = ["sub-test002"]
+    list2 = ["sub-test003"]
+
+    # add subjects to participants.tsv
+    append_list_as_row("testing_data/participants.tsv", list1)
+    append_list_as_row("testing_data/participants.tsv", list2)
+
+
+def test_testing_with_uncertainty():
+    # Test config. Uses Uncertainty
+    subprocess.check_output(["ivadomed -c testing_data/model_config_test.json"], shell=True)
+
+
+def test_training():
+    # Train config 
+    subprocess.check_output(["ivadomed -c testing_data/model_config.json"], shell=True)
+
+
+# def test_training_curve():
+# using the results from previous training
+#   command = "ivadomed_training_curve -i testing_script/ -o training"
+#  subprocess.check_output(command, shell=True)
+
+
+def test_create_eval_json():
+    # modify train config
+    command = "cp testing_data/model_config.json testing_data/model_config_eval.json"
+    subprocess.check_output(command, shell=True)
+    file_conf = open("testing_data/model_config_eval.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_eval.json", "w")
+    initial_config["command"] = "eval"
+    initial_config["transformation"] = {
+        "Resample": {
+            "wspace": 0.75,
+            "hspace": 0.75
+        },
+        "RandomAffine": {
+            "degrees": 4.6,
+            "translate": [0.03, 0.03],
+            "scale": [0.98, 1]
+        },
+        "NumpyToTensor": {},
+        "NormalizeInstance": {
+            "applied_to": ["im"]
+        }}
+    initial_config["testing_parameters"] = {
+        "binarize_prediction": False,
+        "uncertainty": {
+            "epistemic": False,
+            "aleatoric": False,
+            "n_it": 2
+        }}
+    json.dump(initial_config, file_conf)
+
+
+def test_eval():
+    subprocess.check_output(["ivadomed -c testing_data/model_config_eval.json"], shell=True)
+
+
+def test_create_automate_training_json():
+    # modify train config
+    command = "cp testing_data/model_config_eval.json testing_data/model_config_auto.json"
+    subprocess.check_output(command, shell=True)
+    file_conf = open("testing_data/model_config_auto.json", "r")
+    initial_config = json.load(file_conf)
+    file_conf.close()
+    file_conf = open("testing_data/model_config_auto.json", "w")
+    initial_config["command"] = "train"
+    initial_config["gpu"] = "[7]"
+    json.dump(initial_config, file_conf)
+
+
+def test_automate_training():
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --fixed-split"
+    subprocess.check_output(command, shell=True)
+
+    command = "ivadomed_automate_training -c testing_data/model_config_auto.json " \
+              "-p testing_data/hyperparameter_opt.json -n 1 --run-test --all-combin"
```

Comment:
We could add the flag `-t 0.1` for extra coverage :)

Reply:
Right !

Reply:
Done in 
e28665b